service/patch.go

```diff
@@ -86,7 +86,9 @@ func (uis *UIServer) schedulePatchUI(w http.ResponseWriter, r *http.Request) {
 	if err := utility.ReadJSON(utility.NewRequestReader(r), &patchUpdateReq); err != nil {
 		uis.LoggedError(w, r, http.StatusBadRequest, err)
 	}
-	patchUpdateReq.Caller = curUser.Username()
+	if curUser != nil {
+		patchUpdateReq.Caller = curUser.Username()
```

Comment:
I think we should just be returning early at 83, rather than checking this again here.

Reply:
(Legacy about to go anyway anyway haha so i don't care if we fix this though)